### ISSUE TITLE
error on grunt build-assets

### ISSUE THREAD

**Issue by byndstd** [humhub/humhub#4146]
### The production build will compress scripts and stylesheets configured in humhub\assets\AppAsset into the following files:

@humhub/static/js/all-*.js
@humhub/static/css/all-*.css

### ```
Warning: Command failed: rm static/js/humhub-*.js ; rm static/css/humhub-*.css ; rm -rf static/assets/*/ ; cd protected ; php yii asset humhub/config/assets.php humhub/config/assets-prod.php
rm: cannot remove 'static/js/humhub-*.js': No such file or directory
rm: cannot remove 'static/css/humhub-*.css': No such file or directory
Exception 'yii\base\InvalidConfigException' with message 'The directory does not exist: /opt/lampp/htdocs/imb/protected/humhub/config/../../../static/assets'

in /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/web/AssetManager.php:213

Stack trace:
#0 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/BaseObject.php(109): yii\web\AssetManager->init()
#1 [internal function]: yii\base\BaseObject->__construct()
#2 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/di/Container.php(400): ReflectionClass->newInstanceArgs()
#3 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/di/Container.php(159): yii\di\Container->build()
#4 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
#5 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(162): yii\BaseYii::createObject()
#6 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(229): yii\console\controllers\AssetController->getAssetManager()
#7 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(190): yii\console\controllers\AssetController->loadConfiguration()
#8 [internal function]: yii\console\controllers\AssetController->actionCompress()
#9 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#10 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#11 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction()
#12 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction()
#13 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
#14 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction()
#15 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest()
#16 /opt/lampp/htdocs/imb/protected/yii(29): yii\base\Application->run()
#17 {main}
 Use --force to continue.
```


### Additional info

| Q                | A
| ---------------- | ---
| HumHub version   | 1.5.1
| PHP version      |  7.4
| Operating system | linux

**Comment by luke-** [comment]
Works fine with the latest master on my environment.
Which humhub version are you using?

**Comment by byndstd** [comment]
> Works fine with the latest master on my environment.
> Which humhub version are you using?

Varsion humhub-1.5.1 I'm using the github one.

**Comment by luke-** [comment]
Can you please provide steps to reproduce this problem?

**Comment by byndstd** [comment]
> Can you please provide steps to reproduce this problem?
That wat I did exactly
```
lotfi@Lenovo:/opt/lampp/htdocs/imb$ sudo npm install grunt --save-dev
[sudo] password for lotfi:          
+ grunt@1.1.0
updated 1 package and audited 234 packages in 2.549s

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities

lotfi@Lenovo:/opt/lampp/htdocs/imb$ sudo grunt build-assets
Running "shell:buildAssets" (shell) task
Loading configuration from 'humhub/config/assets.php'...
Collecting source bundles information...
Creating output bundle 'app':
  Compressing JavaScript files...
Running "uglify:assets" (uglify) task
[D] Task source: /opt/lampp/htdocs/imb/node_modules/grunt-contrib-uglify/tasks/uglify.js
>> 1 file created 1.24 MB → 570.93 kB

Done.
  JavaScript files compressed into '/opt/lampp/htdocs/imb/protected/humhub/config/../../../static/js/humhub-app.js'.
  Compressing CSS files...
PHP Warning 'yii\base\ErrorException' with message 'file_get_contents(/opt/lampp/htdocs/imb/static/assets/d3daaf45/nprogress.css): failed to open stream: No such file or directory'

in /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php:598

Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleError()
#1 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(598): file_get_contents()
#2 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(546): yii\console\controllers\AssetController->combineCssFiles()
#3 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(373): yii\console\controllers\AssetController->compressCssFiles()
#4 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(199): yii\console\controllers\AssetController->buildTarget()
#5 [internal function]: yii\console\controllers\AssetController->actionCompress()
#6 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#7 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#8 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction()
#9 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction()
#10 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
#11 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction()
#12 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest()
#13 /opt/lampp/htdocs/imb/protected/yii(29): yii\base\Application->run()
#14 {main}
Warning: Command failed: rm static/js/humhub-*.js ; rm static/css/humhub-*.css ; rm -rf static/assets/*/ ; cd protected ; php yii asset humhub/config/assets.php humhub/config/assets-prod.php
PHP Warning 'yii\base\ErrorException' with message 'file_get_contents(/opt/lampp/htdocs/imb/static/assets/d3daaf45/nprogress.css): failed to open stream: No such file or directory'

in /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php:598

Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleError()
#1 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(598): file_get_contents()
#2 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(546): yii\console\controllers\AssetController->combineCssFiles()
#3 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(373): yii\console\controllers\AssetController->compressCssFiles()
#4 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(199): yii\console\controllers\AssetController->buildTarget()
#5 [internal function]: yii\console\controllers\AssetController->actionCompress()
#6 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#7 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
#8 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction()
#9 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction()
#10 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
#11 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction()
#12 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest()
#13 /opt/lampp/htdocs/imb/protected/yii(29): yii\base\Application->run()
#14 {main}
 Use --force to continue.

Aborted due to warnings.
lotfi@Lenovo:/opt/lampp/htdocs/imb$ 

```

**Comment by luke-** [comment]
make sure your installed all composer dependencies before:
https://docs.humhub.org/docs/develop/environment#run-composer

**Comment by byndstd** [comment]
> make sure that you installed all composer dependencies before:
> https://docs.humhub.org/docs/develop/environment#run-composer

Everything is installed before I did `sudo grunt build-assets`

**Comment by luke-** [comment]
can you please post the output of `composer install`

**Comment by byndstd** [comment]
> can you please post the output of `composer install`

```
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package facebook/webdriver is abandoned, you should avoid using it. Use php-webdriver/webdriver instead.
Package zendframework/zend-escaper is abandoned, you should avoid using it. Use laminas/laminas-escaper instead.
Package zendframework/zend-http is abandoned, you should avoid using it. Use laminas/laminas-http instead.
Package zendframework/zend-hydrator is abandoned, you should avoid using it. Use laminas/laminas-hydrator instead.
Package zendframework/zend-ldap is abandoned, you should avoid using it. Use laminas/laminas-ldap instead.
Package zendframework/zend-loader is abandoned, you should avoid using it. Use laminas/laminas-loader instead.
Package zendframework/zend-stdlib is abandoned, you should avoid using it. Use laminas/laminas-stdlib instead.
Package zendframework/zend-uri is abandoned, you should avoid using it. Use laminas/laminas-uri instead.
Package zendframework/zend-validator is abandoned, you should avoid using it. Use laminas/laminas-validator instead.
Generating autoload files
13 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

**Comment by luke-** [comment]
Hmm, following works fine on my system.

```
cd /tmp
git clone git@github.com:humhub/humhub.git
cd humhub
composer install
npm install grunt
grunt build-assets
```

**Comment by byndstd** [comment]
it's seems that I did a new install of 1.5.2 version

**Comment by luke-** [comment]
So anything works with a fresh 1.5.2?

**Comment by boomtow** [comment]
> ### grunt build-assets
> ### The production build will compress scripts and stylesheets configured in humhub\assets\AppAsset into the following files:
> @humhub/static/js/all-_.js @humhub/static/css/all-_.css
> 
> ### ```
> Warning: Command failed: rm static/js/humhub-_.js ; rm static/css/humhub-_.css ; rm -rf static/assets/_/ ; cd protected ; php yii asset humhub/config/assets.php humhub/config/assets-prod.php rm: cannot remove 'static/js/humhub-_.js': No such file or directory
> rm: cannot remove 'static/css/humhub-*.css': No such file or directory
> Exception 'yii\base\InvalidConfigException' with message 'The directory does not exist: /opt/lampp/htdocs/imb/protected/humhub/config/../../../static/assets'
> 
> in /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/web/AssetManager.php:213
> 
> Stack trace:
> #0 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/BaseObject.php(109): yii\web\AssetManager->init()
> #1 [internal function]: yii\base\BaseObject->__construct()
> #2 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/di/Container.php(400): ReflectionClass->newInstanceArgs()
> #3 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/di/Container.php(159): yii\di\Container->build()
> #4 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get()
> #5 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(162): yii\BaseYii::createObject()
> #6 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(229): yii\console\controllers\AssetController->getAssetManager()
> #7 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(190): yii\console\controllers\AssetController->loadConfiguration()
> #8 [internal function]: yii\console\controllers\AssetController->actionCompress()
> #9 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
> #10 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams()
> #11 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction()
> #12 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction()
> #13 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
> #14 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/console/Application.php(147): yii\console\Application->runAction()
> #15 /opt/lampp/htdocs/imb/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest()
> #16 /opt/lampp/htdocs/imb/protected/yii(29): yii\base\Application->run()
> #17 {main}
> Use --force to continue.
> 
> ```
> 
> 
> ### Additional info
> 
> | Q                | A
> | ---------------- | ---
> | HumHub version   | 1.5.1
> | PHP version      |  7.4
> | Operating system | linux
> ```

1.5.2

**Comment by boomtow** [comment]
### Additional info

| Q                | A
| ---------------- | ---
| HumHub version   | 1.5.1
| HumHub version   | 1.?
| PHP version      |  7.4
| Operating system | linux

**Comment by luke-** [comment]
Sorry I cannot reproduce. 

The grunt build seems running on the latest master:
https://github.com/humhub/humhub/issues/4146#issuecomment-635857660

Seems this is related to your local installation.

**Comment by armezit** [comment]
I'm also facing this issue after upgrading from 1.5.0 to 1.5.1

**Comment by luke-** [comment]
@armezit how did you update?

**Comment by buddh4** [comment]
```
Exception 'yii\base\InvalidConfigException' with message 'The directory does not exist: /opt/lampp/htdocs/imb/protected/humhub/config/../../../static/assets'
```

Is there a `/opt/lampp/htdocs/imb/static/assets` directory in your installation ?

**Comment by armezit** [comment]
@luke- I updated manually (git based installation). 

This is the output of `grunt build-assets`:

```
Running "shell:buildAssets" (shell) task
rm: can't remove 'static/css/humhub-*.css': No such file or directory
Loading configuration from 'humhub/config/assets.php'...
Collecting source bundles information...
Creating output bundle 'app':
  Compressing JavaScript files...
Running "uglify:assets" (uglify) task
[D] Task source: /var/www/htdocs/node_modules/grunt-contrib-uglify/tasks/uglify.js
>> 1 file created 1.23 MB → 570.93 kB

Done.
  JavaScript files compressed into '/var/www/htdocs/protected/humhub/config/../../../static/js/humhub-app.js'.
  Compressing CSS files...
PHP Warning 'yii\base\ErrorException' with message 'file_get_contents(/var/www/htdocs/static/assets/aee6c344/nprogress.css): failed to open stream: No such file or directory'

in /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php:598

Stack trace:
#0 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(598): yii\console\ErrorHandler->handleError('???', '???', '???', '???', '???')
#1 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(598): ::file_get_contents('???')
#2 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(546): yii\console\controllers\AssetController->combineCssFiles('???', '???')
#3 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(373): yii\console\controllers\AssetController->compressCssFiles('???', '???')
#4 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/controllers/AssetController.php(199): yii\console\controllers\AssetController->buildTarget('???', '???', '???')
#5 /var/www/htdocs/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): yii\console\controllers\AssetController->actionCompress('???', '???')
#6 /var/www/htdocs/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): ::call_user_func_array:{/var/www/htdocs/protected/vendor/yiisoft/yii2/base/InlineAction.php:57}('???', '???')
#7 /var/www/htdocs/protected/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams('???')
#8 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/Controller.php(164): yii\console\controllers\AssetController->runAction('???', '???')
#9 /var/www/htdocs/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\console\controllers\AssetController->runAction('???', '???')
#10 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/Application.php(180): humhub\components\console\Application->runAction('???', '???')
#11 /var/www/htdocs/protected/vendor/yiisoft/yii2/console/Application.php(147): humhub\components\console\Application->runAction('???', '???')
#12 /var/www/htdocs/protected/vendor/yiisoft/yii2/base/Application.php(386): humhub\components\console\Application->handleRequest('???')
#13 /var/www/htdocs/protected/yii(29): humhub\components\console\Application->run()
#14 {main}
```

The `<web_root>/static/assets` directory exists and all other assets works fine but the `nprogress`.

Here is the contents of the `<web_root>/static/assets` directory:
```
$ ls static/assets/
13feb338  1cca5ddc  27931542  2f2cc1b0  3df4d50e  43ee823   58ab0ad9  5992670b  5c4a85b3  72b4cfd  906f8789  975429a0  9c5009d5  b09da488  d0e9904d  dd4ed86c 1b00fca2  25b33a56  2e17db38  36dd6908  403776e1  4b022daf  59553067 5b51b6df  5f6367c1  818d35a  91d6285f  981d411c  a7318dce  c7218603  d8a1e876  fcf697a6
```
All assets has been built, but there's no directory named `aee6c344`.

**Comment by buddh4** [comment]
Can you compare the following files with yours:

- https://github.com/humhub/humhub/blob/master/protected/humhub/assets/NProgressStyleAsset.php
- https://github.com/humhub/humhub/blob/master/protected/humhub/assets/NProgressAsset.php

This Asset was split in 1.5, perhaps, do you think the problem is that both assets point to the same directory @luke- ?

**Comment by buddh4** [comment]
Also make sure that `<web_root>/protected/vendor/npm-asset/nprogress` contains `nprogress.css` and  `nprogress.js`.

**Comment by armezit** [comment]
> Can you compare the following files with yours:
> 
>     * https://github.com/humhub/humhub/blob/master/protected/humhub/assets/NProgressStyleAsset.php
> 
>     * https://github.com/humhub/humhub/blob/master/protected/humhub/assets/NProgressAsset.php
> 

Both files are the same as mine.

**Comment by armezit** [comment]
> Also make sure that `<web_root>/protected/vendor/npm-asset/nprogress` contains `nprogress.css` and `nprogress.js`.

```
$ ls <web_root>/protected/vendor/npm-asset/nprogress/dist
nprogress.js  nprogress.js.map  nprogress.mjs  nprogress.mjs.map  nprogress.umd.js  nprogress.umd.js.map
```
and
```
$ ls <web_root>/protected/vendor/npm-asset/nprogress/css
nprogress.css  nprogress.scss
```

Shouldn't `nprogress.css` be in the `dist` directory?

**Comment by buddh4** [comment]
Did you run `composer install` after the update? I don't think the nprogress library was updated between 1.4 and 1.5 but it seems you have a different package than me:

![grafik](https://user-images.githubusercontent.com/9800854/85875458-81fcce80-b7d4-11ea-9006-d566269ac932.png)

Maybe, delete the `<web_root>/protected/vendor/npm-asset/nprogress` and run `composer install`.

**Comment by armezit** [comment]
> I assume you've got the `1.0.0-patch0` or `1.0.0-patch1` from https://asset-packagist.org/package/npm-asset/nprogress but I'am not sure why.

@buddh4  You are right. The  `1.0.0-patch1` has been installed.

In `1.5.1`, the version constraint of the `npm-asset/nprogress` is [`"*"`](https://github.com/humhub/humhub/blob/a7076084ede1c11ca43fedbac296c11a12593a5a/composer.json#L42)

In `1.5.2`, it's ["^0.2"](https://github.com/humhub/humhub/blob/7635a25f81280e6cb32ebf805ef25b97e81b18e5/composer.json#L42)

Maybe this is why the `1.0.0-patch1` has been installed.

**Comment by armezit** [comment]
After upgrading to humhub 1.5.2, all assets built successfully.

I think version constraint of the `npm-asset/nprogress` in 1.5.1 was causing error.

**Comment by luke-** [comment]
So this commit in v1.5.2 fixes this issue? https://github.com/humhub/humhub/commit/53a439d1f82b57b589b3e25ff5de0538811b5fe8#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780 

(I'm not sure why I added the version constraint to NProgress.)

**Comment by armezit** [comment]
> So this commit in v1.5.2 fixes this issue? [53a439d#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780](https://github.com/humhub/humhub/commit/53a439d1f82b57b589b3e25ff5de0538811b5fe8#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780)
> 
> (I'm not sure why I added the version constraint to NProgress.)

That's right. I just changed `"*"` to `"^0.2"`. after `composer install` assets are building correctly.

**Comment by wannerdev** [comment]
> > So this commit in v1.5.2 fixes this issue? [53a439d#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780](https://github.com/humhub/humhub/commit/53a439d1f82b57b589b3e25ff5de0538811b5fe8#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780)
> > (I'm not sure why I added the version constraint to NProgress.)
> 
> That's right. I just changed `"*"` to `"^0.2"`. after `composer install` assets are building correctly.

I had the same issue with nprogress and changing it from `"*"` to `"^0.2"` solved it.  
Afterwards though i had the same problem with the clipboard-polyfile library.
I changed that version from `"*"` to `"^2.8.6"` since then i have no more asset errors in my console.
I am running humhub 1.4.4 just fyi.
